sklearn/multioutput.py

```diff
@@ -0,0 +1,212 @@
+from sklearn.utils import check_random_state
+from sklearn.utils.validation import _num_samples
+from sklearn.utils.validation import check_consistent_length
+from sklearn.utils.validation import check_is_fitted
+from sklearn.externals.joblib import Parallel
+from sklearn.externals.joblib import delayed
+
+
+from sklearn.multiclass import OneVsRestClassifier
+
+
+class MultiOneVsRestClassifier(BaseEstimator):
+    """
+    Converts any classifer estimator into 
+    a multi-target classifier estimator. 
+
+    This class fits and predicts a series of one-versus-all models 
+    to response matrix Y, which has n_samples and p_target variables,
+    on the predictor Matrix X with n_samples and m_feature variables. 
+    This allows for multiple target variable classifications. For each 
+    target variable (column in Y), a separate OneVsRestClassifier is fit. 
+    See the base OneVsRestClassifier Class in sklearn.multiclass 
+    for more details.
+    
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit` &
+         `predict_proba`.
+    n_jobs : int, optional, default: 1
+        The number of jobs to use for the computation. 
+        If -1 all CPUs are used.
+        If 1 is given, no parallel computing code is used at all, 
+        which is useful for debugging. For n_jobs below -1, 
+        (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        
+        Note that parallel processing only occurs if there is
+        multiple classes within each target variable. 
+        It does each target variable in y in series.
+
+    Attributes
+    __________
+    estimator: Sklearn estimator: The base estimator used to constructe
+    the model. 
+        
+    """
+    
+    def __init__(self, estimator=None, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data. 
+        Creates a seperate model for each Response column. 
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : (sparse) array-like, shape = [n_samples, p_targets]
+            Multi-class targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+
+        # check to see that the data is numeric 
+
+        # check to see that the X and y have the same number of rows. 
+
+        # Calculate the number of classifiers
+        self._num_y = y.shape[1]
+
+        ## create a dictionary to hold the estimators. 
+        self.estimators_ ={}
+        
+        for i in range(self._num_y):
+            # init a new classifer for each and fit it. 
+            estimator = clone(self.estimator)   #make a fresh clone
+            ovr = OneVsRestClassifier(estimator,self.n_jobs)
```

Comment:
umm. I think we should train the `n_outputs` using the base estimator in parallel and not use `OneVsRestClassifier` implicitly. If the base estimator supports multiclass classification, well and good, otherwise we should just error.